bin/sozo/src/commands/execute.rs

```diff
@@ -82,43 +130,52 @@ impl ExecuteArgs {
             )
             .await?;
 
-            let contract_address = match &descriptor {
-                ResourceDescriptor::Address(address) => Some(*address),
-                ResourceDescriptor::Tag(tag) => {
-                    let selector = naming::compute_selector_from_tag(tag);
-                    world_diff.get_contract_address(selector)
-                }
-                ResourceDescriptor::Name(_) => {
-                    unimplemented!("Expected to be a resolved tag with default namespace.")
+            let call_args_list: Vec<CallArgs> = self.calls.iter()
+                .map(|s| CallArgs::from_string(s))
+                .collect::<Result<Vec<_>>>()?;
+
+            for call_args in call_args_list {
+                let descriptor = call_args.tag_or_address.ensure_namespace(&profile_config.namespace.default);
+
+                let contract_address = match &descriptor {
+                    ResourceDescriptor::Address(address) => Some(*address),
+                    ResourceDescriptor::Tag(tag) => {
+                        let selector = naming::compute_selector_from_tag(tag);
+                        world_diff.get_contract_address(selector)
+                    }
+                    ResourceDescriptor::Name(_) => {
+                        unimplemented!("Expected to be a resolved tag with default namespace.")
+                    }
                 }
+                .ok_or_else(|| anyhow!("Contract {descriptor} not found in the world diff."))?;
+
+                trace!(
+                    contract=?descriptor,
+                    entrypoint=call_args.entrypoint,
+                    calldata=?call_args.calldata,
+                    "Executing Execute command."
+                );
+
+                let calldata = if let Some(cd) = call_args.calldata {
+                    calldata_decoder::decode_calldata(&cd)?
+                } else {
+                    vec![]
+                };
+
+                let call = Call {
+                    calldata,
+                    to: contract_address,
+                    selector: snutils::get_selector_from_name(&call_args.entrypoint)?,
+                };
+
+                let invoker = Invoker::new(&account, txn_config);
```

Comment:
You're not doing a multi call if you invoke the call at each iteration.
Instead, create the `Invoker` before starting the loop. Then use the `Invoker.add_call`. Finally, use the `Invoker` to multi call.